library/include/G4CMPEmpiricalLindhardNIEL.hh

```diff
@@ -0,0 +1,75 @@
+/***********************************************************************\
+ * This software is licensed under the terms of the GNU General Public *
+ * License version 3 or later. See G4CMP/LICENSE for the full license. *
+\***********************************************************************/
+
+/// \file library/include/G4CMPEmpiricalLindhardNIEL.hh
+/// \brief NIEL calculation from Ionization yield measurement 
+/// in a germanium CDMSlite detector using photo-neutron sources.
+///
+/// Computation of NIEL according to Lindhard (Lewin Smith) yield with 
+/// a parameter k that depends on the energy.
+///
+/// Paper DOI: https://doi.org/10.1103/PhysRevD.105.122002
+//
+// 20250212  David Sadek
+
+#ifndef G4CMPEmpiricalLindhardNIEL_hh
+#define G4CMPEmpiricalLindhardNIEL_hh 1
+
+#include "G4VNIELPartition.hh"
+#include "G4CMPLewinSmithNIEL.hh"
+#include "G4SystemOfUnits.hh"
+#include <cfloat>
+
+
+class G4CMPEmpiricalLindhardNIEL : public G4CMPLewinSmithNIEL {
+public:
+    // Default constructor
+    G4CMPEmpiricalLindhardNIEL();
+
+    virtual ~G4CMPEmpiricalLindhardNIEL() override {;}
+
+  
+  // return the fraction of the specified energy which will be deposited as NIEL
+  // if an incoming particle is stopped in the specified material,
+  // energy in native G4 energy units.
+  //
+  // The parameter k is a function of energy
+
+  // Setters
+  void SetEmpiricalklow(G4double kl);
+  void SetEmpiricalkhigh(G4double kh);
+  void SetEmpiricalElow(G4double el);
+  void SetEmpiricalEhigh(G4double eh);
+  void SetEmpiricalkFixed(G4double fixedK);
+  void SetEmpiricalEnergyDependentK(bool use);
```

Comment:
All of these functions could and should be inlined directly here.  There is no reason to defer them to the .cc file.